src/Sidekick/App.xaml.cs

```diff
@@ -164,5 +172,33 @@ private void EnsureSingleInstance()
                 Current.Shutdown();
             }
         }
+
+        private void AttachErrorHandlers()
+        {
+            AppDomain.CurrentDomain.UnhandledException += (s, e) =>
+            {
+                var exception = (Exception)e.ExceptionObject;
+                LogUnhandledException(exception);
+            };
+
+            DispatcherUnhandledException += (s, e) =>
+            {
+                LogUnhandledException(e.Exception);
+                e.Handled = true;
+            };
+
+            TaskScheduler.UnobservedTaskException += (s, e) =>
+            {
+                LogUnhandledException(e.Exception);
+                e.SetObserved();
+            };
+        }
+
+        private void LogUnhandledException(Exception ex)
+        {
+            logger.Fatal(ex, "Unhandled exception in application root");
+            AdonisUI.Controls.MessageBox.Show(ApplicationResources.FatalErrorOccured, buttons: AdonisUI.Controls.MessageBoxButton.OK);
```

Comment:
My first draft was also to shutdown the app, but I think that showing a message box right in front of a player while Sidekick crashes can be very dangerous if he's in hardcore for example.